### PR TITLE
doc/security.md: add notes for proxy security

### PR DESCRIPTION
**Pull request by xiang90**
Fix https://github.com/coreos/etcd/issues/3599

/cc @joshix @heyitsanthony @philips

### CODE REVIEW

File: Documentation/security.md [hunk 1]
```diff
@@ -143,6 +143,14 @@ $ etcd -name infra2 -data-dir infra2 \
 
 The etcd members will form a cluster and all communication between members in the cluster will be encrypted and authenticated using the client certificates. You will see in the output of etcd that the addresses it connects to use HTTPS.
 
+## Notes For etcd Proxy
+
+etcd proxy terminates the TLS from its client if the connection is secure, and uses proxy's own key/cert specified in `--peer-key-file` and `--peer-cert-file` to communicate with etcd members.
+
+Proxy communicates with etcd member through both advertise client urls and advertise peer urls of that member. It forwards the client requests to etcd members’ advertise client urls, and it syncs the initial cluster configuration through etcd members’ advertise peer urls.
```
Comment (joshix) [inline Documentation/security.md]: "The proxy communicates with etcd members through both the `--advertise-client-urls` and `--advertise-peer-urls` of a given member. It forwards client requests to etcd members’ advertised client urls, and it syncs the initial cluster configuration through etcd members’ advertised peer urls."

File: Documentation/security.md [hunk 2]
```diff
@@ -143,6 +143,14 @@ $ etcd -name infra2 -data-dir infra2 \
...
+
+So when the client authentication is enabled for etcd member, user must ensure that the peer cert specified (`--peer-cert-file`) for the proxy is valid for client authentication. Also when the peer authentication is enabled, user must ensure that the peer cert specified for proxy is valid for peer authentication.
```
Comment (joshix) [inline Documentation/security.md]: "When client authentication is enabled for an etcd member, the administrator must ensure that the peer certificate specified in the proxy's `--peer-cert-file` option is valid for that authentication. The proxy's peer certificate must also be valid for peer authentication if peer authentication is enabled."

**Comment by xiang90** [comment]
@joshix All fixed. PTAL!

File: Documentation/security.md [hunk 3]
```diff
@@ -143,6 +143,14 @@ $ etcd -name infra2 -data-dir infra2 \
 
 The etcd members will form a cluster and all communication between members in the cluster will be encrypted and authenticated using the client certificates. You will see in the output of etcd that the addresses it connects to use HTTPS.
 
+## Notes For etcd Proxy
+
+etcd proxy terminates the TLS from its client if the connection is secure, and uses proxy's own key/cert specified in `--peer-key-file` and `--peer-cert-file` to communicate with etcd members.
+
+The proxy communicates with etcd members through both the --advertise-client-urls and --advertise-peer-urls of a given member. It forwards client requests to etcd members’ advertised client urls, and it syncs the initial cluster configuration through etcd members’ advertised peer urls.
```
Comment (joshix) [inline Documentation/security.md]: monospace for the options: `--advertise-client-urls`, `--advertise-peer-urls` (with backticks ``)

File: Documentation/security.md [hunk 4]
```diff
@@ -143,6 +143,14 @@ $ etcd -name infra2 -data-dir infra2 \
...
+
+When client authentication is enabled for an etcd member, the administrator must ensure that the peer certificate specified in the proxy's --peer-cert-file option is valid for that authentication. The proxy's peer certificate must also be valid for peer authentication if peer authentication is enabled.
```
Comment (joshix) [inline Documentation/security.md]: monospace for `--peer-cert-file`

**Comment by xiang90** [comment]
@joshix Fixed.

**Comment by joshix** [comment]
LGTM

**Comment by xiang90** [comment]
@joshix Thanks for reviewing this!

**Comment by roboll** [comment]
@xiang90 thanks for the heads up.

**Pull request merged**